internal/database/actions_test.go

```diff
@@ -3,14 +3,46 @@ package database
 import (
 	"database/sql"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"os"
 	"testing"
 
-	"github.com/ooni/probe-engine/model"
 	db "upper.io/db.v3"
 )
 
+type locationInfo struct {
+	asn         uint
+	countryCode string
+	ip          string
+	networkName string
+	resolverIP  string
+}
```

Comment:
This implements the `util.LocationProvider` interface and allows to write mocked tests.